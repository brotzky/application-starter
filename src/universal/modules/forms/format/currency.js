import numeral from 'numeral';

const currencyFormat = currency => numeral(currency).format('$ 0,0.00');

export default currencyFormat;
