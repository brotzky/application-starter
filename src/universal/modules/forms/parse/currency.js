export default function(value) {
  const stripNonDigits = value && value.replace(/[^0-9\.]+/g, '');
  return stripNonDigits ? parseFloat(stripNonDigits) : '';
}
