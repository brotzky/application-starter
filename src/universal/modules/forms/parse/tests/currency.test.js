import currency from '../currency';

describe('[Form Field Parsing] currency', () => {
  describe('currency()', () => {
    it('should strip a currency string formatting and return a number', () => {
      expect(currency('$1,000.00')).toMatchSnapshot();
    });
  });
});
