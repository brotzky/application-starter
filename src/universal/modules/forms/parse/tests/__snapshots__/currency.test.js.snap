// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form Field Parsing] currency currency() should strip a currency string formatting and return a number 1`] = `1000`;
